hazelcast/src/main/java/com/hazelcast/client/impl/clientside/HazelcastClientInstanceImpl.java

```diff
@@ -421,6 +430,21 @@ public void start() {
         }
     }
 
+    private void startHeartbeat() {
+        long heartbeatTimeout = properties.getPositiveMillisOrDefault(HEARTBEAT_TIMEOUT);
+        long heartbeatInterval = properties.getPositiveMillisOrDefault(HEARTBEAT_INTERVAL);
+        ILogger logger = loggingService.getLogger(HeartbeatManager.class);
+        HeartbeatManager.start(this, executionService, logger,
+                heartbeatInterval, heartbeatTimeout,
+                Collections.unmodifiableCollection(connectionManager.getActiveConnections()));
```

Comment:
What happens when a new member joins? I think we have to fetch the active connections, each time we are about to ping

Reply:
getActiveConnections does not return copy. The modifications on the map are visible in the returned Collections. So there is no need for asking again to the connection manager.

I see that this is already confusing and may be a good sign to revert the second commit.